lib/model.js

```diff
@@ -1427,6 +1427,9 @@ Model.diffIndexes = function diffIndexes(options, callback) {
   return this.db.base._promiseOrCallback(callback, cb => {
     cb = this.$wrapCallback(cb);
     this.listIndexes((err, indexes) => {
+      if (err) {
```

Comment:
We definitely need an error check here, thanks for adding. But in the case of `NamespaceNotFound`, we know that there are no indexes in the db because the collection doesn't exist. So I think we should add a check `if (err.codeName === 'NamespaceNotFound') { indexes = []; } else { return cb(err); }`

Reply:
Perfect, that's exactly the behavior I'd expect when catching that error, makes sense to just implement it in the library itself. Will do!